src/common/os/win32/guid.cpp

```diff
@@ -76,9 +75,17 @@ void GenerateRandomBytes(void* buffer, FB_SIZE_T size)
 
 void GenerateGuid(Guid* guid)
 {
-	const HRESULT error = CoCreateGuid((GUID*) guid);
-	if (!SUCCEEDED(error))
-		Firebird::system_call_failed::raise("CoCreateGuid", error);
+	// Use UuidCreateSequentional() because is has stronger protection
+	// from collisions.
+	const RPC_STATUS error = UuidCreateSequential(guid);
```

Comment:
No. They generate different versions of UUID. CoCreateGuid - version 4, UuidCreateSequential - version 1.

Reply:
But in CORE-3238 we had explicitly agreed to generate version 4 UUIDs. It should not be broken without prior discussion.

Reply:
In the ticket I see proposal but not reasoning for it. AFAIU version 4 was suggested because it is the simplest. On the other hand it is completely bad for indexing and have higher collision probability.

Reply:
I suppose the reason is consistency (Linux UUIDs are version 4 by design, thus it makes sense to have Windows UUIDs of the same version).

Reply:
I personally see no point in this. UUID must be unique, consistency or random bytes is a strange thing.